tests/webidl/test.idl

```diff
@@ -30,8 +30,8 @@ interface Child2 {
   void virtualFunc2();
   void virtualFunc3(long x);
   void virtualFunc4(long x);
-  static void runVirtualFunc(Child2 self);
-  static void runVirtualFunc3(Child2 self, long x);
+  static void runVirtualFunc(Child2 myself);
+  static void runVirtualFunc3(Child2 myself, long x);
```

Comment:
is this change just aesthetic? or is there something deeper?

Reply:
Sort of. When generating the cpp side of the bindings class methods take an argument that refers to the object that called it. This variable is named 'self', so this change means that functions defined in IDL now cannot have a argument named self, since that would clash with the automatically added one.

So `void runVirtualFunc(Child2 myself);` becomes something like this in C++: 
```
void EMSCRIPTEN_KEEPALIVE emscripten_bind_Child2_get_runVirtualFunc_1(Child2* self, Child2 myself) {
  return self->runVirtualFunc(myself);
}
```

Reply:
Interesting, thanks. So this PR does come with some risk of breakage for people.

How about changing the internally-used name to something less risky, like `__webidl_self` or such?

Reply:
Yeah that's true. I would think most people would avoid naming arguments things like 'self', since that's a keyword in heaps of languages (Rust, Python, Objective-C, Swift, ...), but the fact that they were used here means that I was wrong.

Personally I think that `__webidl_self` isn't a super clear name, how about just `_self` or `__self`?

On the other hand, @wingo's changes are also breaking, so if both of these are in the same release users would have to update their IDL anyway.

But at the end of the day I'm not too fussed, I'm happy to do whatever you prefer.

Reply:
Fair points. Yeah, how about `__self`? Seems like a reasonable choice.

Reply:
OK, I took a quick look and internally 'self' is used in a number of different places. It would take some time to implement and test this, so I think it makes more sense as a separate PR. This change would also conflict a fair bit with @wingo's changes (#8849), so I think that change should be made after his change is merged.

Reply:
I see, thanks.

I'd be ok to land this as is if it doesn't conflict with @wingo's changes - I guess we can consider renaming `self` to something else if any issues come up later.

Reply:
Pretty sure this won't conflict. I've created an issue for the follow up work here #8985 so I don't forget.